sound/soc/sof/imx/imx8.c

```diff
@@ -116,7 +117,16 @@ static void imx8_dsp_handle_request(struct imx_dsp_ipc *ipc)
 {
 	struct imx8_priv *priv = imx_dsp_get_data(ipc);
 
-	snd_sof_ipc_msgs_rx(priv->sdev);
+	int p; /* panic code */
+
+	/* Read the message from the debug box. */
+	sof_mailbox_read(priv->sdev, priv->sdev->debug_box.offset + 4, &p, 4);
```

Comment:
The panic code probably has a more well defined type than "int". Maybe have "u32 p" or "s32 p" and then use "sizeof(p)" as sof_mailbox_read argument?